src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -3376,7 +3389,7 @@ private string[] GetUniqueKeyPropertyNames(string[] suppliedPropertyNames)
 					}
 					else
 					{
-						Check(session.Batcher.ExecuteNonQuery(statement), tableId, j, expectation, statement);
+						Check(session.Batcher.ExecuteNonQuery(statement), tableId, j, expectation, statement, IsPropertyBasedOptimisticLocking(loadedState));
```

Comment:
`IsPropertyBasedOptimisticLocking(loadedState)` is always `false` at this stage, I believe.

Reply:
Nope.